spec/migrations/20190617191109_move_awx_credentials_to_authentications_spec.rb

```diff
@@ -0,0 +1,209 @@
+          :type        => "ManageIQ::Providers::EmbeddedAnsible::AutomationManager::MachineCredential",
+          :manager_ref => "3",
+          :options     => {
+            :ssh_key_data    => "$encrypted$",
+            :become_username => "machine",
+            :become_password => "$encrypted$"
+          }
+        )
+
+        stub_awx_cred_for_id(awx_conn, "3", [])
+
+        migrate
+
+        expect(auth.reload.options).to be_nil
+      end
+
+      it "migrates options fields from the awx database" do
+        auth = authentication.create!(
+          :name        => "machine",
+          :userid      => "machine",
+          :type        => "ManageIQ::Providers::EmbeddedAnsible::AutomationManager::MachineCredential",
+          :manager_ref => "3",
+          :options     => {
+            :ssh_key_data    => "$encrypted$"
+          }
+        )
+
+        awx_credential = {
+          "inputs" => {
+            "authorize"     => false,
+            "become_method" => "sudo",
+            "client"        => "someclient",
+            "domain"        => "thebestdomain",
+            "host"          => "host.example.com",
+            "project"       => "manageiq",
+            "subscription"  => "sub",
+            "tenant"        => "atenant"
+          }.to_json
+        }
+
+        stub_awx_cred_for_id(awx_conn, "3", [awx_credential])
+
+        migrate
+        auth.reload
+
+        expected_options = {
+          :authorize     => false,
+          :become_method => "sudo",
+          :client        => "someclient",
+          :domain        => "thebestdomain",
+          :host          => "host.example.com",
+          :project       => "manageiq",
+          :subscription  => "sub",
+          :tenant        => "atenant"
+        }
+
+        expect(auth.options).to eq(expected_options)
+      end
+
+      it "migrates real data" do
+        # load the initial set of authentication records
+        auths = YAML.load_file(data_dir.join("initial_vmdb_authenticaions.yaml"))
+        auths.each { |attrs| authentication.create!(attrs) }
+
+        stub_awx_credentials(awx_conn)
+        expect_credential_decrypts
+
+        migrate
+
+        expect_authentications_migrated
+      end
+    end
+  end
+
+  def stub_awx_cred_for_id(connection, id, data)
+    expect(connection).to receive(:async_exec).with("SELECT inputs FROM main_credential WHERE id = $1::BIGINT", [id]).and_return(data)
+  end
+
+  def stub_awx_credentials(connection)
+    credentials = YAML.load_file(data_dir.join("credential_attributes.yaml"))
+    credentials.each do |cred|
+      data = cred["awx_attributes"].to_json
+      stub_awx_cred_for_id(connection, cred["id"], [{"inputs" => data}])
+    end
+  end
+
+  def expect_credential_decrypts
+    stubs = YAML.load_file(data_dir.join("awesome_stubs.yaml"))
+    stubs.each do |h|
+      result = AwesomeSpawn::CommandResult.new("python3 standalone_decrypt.py", h[:output], "", 0)
```

Comment:
I thought awesome_spawn had helper methods for this.

Reply:
See https://github.com/ManageIQ/awesome_spawn/blob/master/lib/awesome_spawn/spec_helper.rb - does that solve it?

Reply:
Example usage from the bot: https://github.com/ManageIQ/miq_bot/blob/master/spec/lib/minigit_service_spec.rb#L23-L24

Reply:
So a few things there...

https://github.com/ManageIQ/awesome_spawn/blob/master/lib/awesome_spawn/spec_helper.rb#L14-L20 just looks broken.

And the bot just vendors the file https://github.com/ManageIQ/miq_bot/blob/master/spec/support/awesome_spawn_helper.rb

So I can go ahead and fix that file up, but I'm not sure if we want to hold this for a new version of awesome spawn.

Reply:
Also even with this (I did the same thing as the bot and copied the file into `spec/support`) I would need to build out the path to the python script to get the stub right rather than using `an_array_including` plus add env support to the existing awesome spawn spec helper.

Reply:
Okay, opened a PR to fix the helper https://github.com/ManageIQ/awesome_spawn/pull/46